installer/cli/checkpoint.py

```diff
@@ -0,0 +1,84 @@
+import os
+import pickle
+from datetime import datetime
+from vm_providers.factory import get_provider_for
+
+
+CHECKPOINT_FILE = '.microk8s.checkpoint'
+CHECKPOINT_PATH = os.path.join(os.path.expanduser('~'), CHECKPOINT_FILE)
+ALIAS_FILE = '.microk8s.rc'
+ALIAS_PATH = os.path.join(os.path.expanduser('~'), ALIAS_FILE)
+
+
+def sync(force: bool = False) -> None:
+    """
+    Synchronise the alias file with the available commands.
+
+    :return: None
+    """
+    if not _check_checkpoint() or force:
+        _sync_commands()
+        _make_checkpoint()
+
+
+def _sync_commands() -> None:
+    """
+    Rewrite the alias file.
+
+    :return: None
+    """
+    if platform in ['linux', 'darwin']:
+        with open(ALIAS_PATH, 'w') as f:
+            for i in _get_microk8s_commands():
+                f.write('alias microk8s.{0}="microk8s {0}"'.format(i))
```

Comment:
Will this write the same aliases as many times as it is called? Do we clean the aliases?

Reply:
This will write all aliases in 1 file (the `open` is above the `for`).  Because the `open` is in write mode, rather than append, the file will be overwritten each time.